src/Containers/vectorized_product_iterator.jl

```diff
@@ -1,10 +1,16 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
 """
     struct VectorizedProductIterator{T}
         prod::Iterators.ProductIterator{T}
     end
 
-Same as `Base.Iterators.ProductIterator` except that it is independent
-on the `IteratorSize` of the elements of `prod.iterators`.
+Cartesian product of the iterators `prod.iterators`. It is the same iterator as
+`Base.Iterators.ProductIterator` except that it is independent of the
+`IteratorSize` of the elements of `prod.iterators`.
```

Comment:
I think this could benefit from a longer comment about why the behavior of `Base.Iterators.ProductIterator` wasn't sufficient and required a wrapper. (Maybe outside the docstring.) What breaks if we use `Base.Iterators.ProductIterator`?

Reply:
In fact, every test passed with `Iterators.ProductIterator` but
https://github.com/JuliaOpt/JuMP.jl/blob/1a4f663504606464292caf2db2d60acba6991af3/examples/cutting_stock_column_generation.jl#L149
fails.
Now I have added tests in `test/Containers/` covering this feature.